worker/src/queue.rs

```diff
@@ -179,15 +342,343 @@ impl AsRef<JsValue> for Queue {
+    V8,
+}
+
+impl Serialize for QueueContentType {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(match self {
+            Self::Json => "json",
+            Self::Text => "text",
+            Self::V8 => "v8",
+        })
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct QueueSendOptions {
+    content_type: Option<QueueContentType>,
+    delay_seconds: Option<u32>,
+}
+
+pub struct MessageBuilder<T> {
+    message: T,
+    delay_seconds: Option<u32>,
+    content_type: QueueContentType,
+}
+
+impl<T: Serialize> MessageBuilder<T> {
+    /// Creates a new message builder. The message must be `serializable`.
+    pub fn message(message: T) -> Self {
+        Self {
+            message,
+            delay_seconds: None,
+            content_type: QueueContentType::Json,
+        }
+    }
+
+    #[must_use]
+    /// The number of seconds to delay a message for within the queue, before it can be delivered to a consumer
+    pub fn with_delay_seconds(mut self, delay_seconds: u32) -> Self {
+        self.delay_seconds = Some(delay_seconds);
+        self
+    }
+
+    #[must_use]
+    /// The content type of the message.
+    /// Default is `QueueContentType::Json`.
+    pub fn with_content_type(mut self, content_type: QueueContentType) -> Self {
+        self.content_type = content_type;
+        self
+    }
+
+    /// Build the message.
+    pub fn build(self) -> SendMessage<T> {
+        SendMessage {
+            message: self.message,
+            options: Some(QueueSendOptions {
+                content_type: Some(self.content_type),
+                delay_seconds: self.delay_seconds,
+            }),
+        }
+    }
+}
+
+pub struct RawMessageBuilder {
+    message: JsValue,
+    delay_seconds: Option<u32>,
+}
+
+impl RawMessageBuilder {
+    /// Creates a new raw message builder. The message must be a `JsValue`.
+    pub fn message(message: JsValue) -> Self {
+        Self {
+            message,
+            delay_seconds: None,
+        }
+    }
+
+    #[must_use]
+    /// The number of seconds to delay a message for within the queue, before it can be delivered to a consumer
+    pub fn with_delay_seconds(mut self, delay_seconds: u32) -> Self {
+        self.delay_seconds = Some(delay_seconds);
+        self
+    }
+
+    /// Build the message with a content type.
+    pub fn build_with_content_type(self, content_type: QueueContentType) -> SendMessage<JsValue> {
+        SendMessage {
+            message: self.message,
+            options: Some(QueueSendOptions {
+                content_type: Some(content_type),
+                delay_seconds: self.delay_seconds,
+            }),
+        }
+    }
+}
+
+pub struct SendMessage<T> {
```

Comment:
The existence of this type might be confusing, could we add documentation? I'm also a little bit worried by the fact that this could either be a `JsValue` or `T`.

Reply:
The intention here is for this to be a mostly internal type, it needs to be `pub` to be in the function signatures but users can't construct it directly as the fields are private.

I've added doc to clarify this.